### PR TITLE
ft(userProfilePage):User can view and edit their profile[#187364815]

### PR DESCRIPTION
**Pull request by Ke-bean**
## ℹ️ Description

Users should be able to amend/edit their profiles depending on a change of preference
 to the entire disclosure of the individual user

## 🧪 How can this be tested?

This can be tested by accessing the Authenticated user dashboard and view && edit profile as needed

## 📸 Screenshots

Screen Shot 2024-07-09 at 16 49 00
Screen Shot 2024-07-09 at 16 49 07

## 📍 Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have created a loom video for the new feature or enhancement
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/authcomponents/editProfile.tsx [hunk 1]
```diff
@@ -1,120 +1,313 @@
...
+
+  const handleSubmit = async (e:React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    try {
+      await updateProfile(profileData).unwrap();
+      await refetch();
+      setIsEditProfile(false);
+    } catch (err) {
+      console.error('Failed to update profile:', err);
+    }
```
Comment (Oliviier-dev) [inline src/components/authcomponents/editProfile.tsx]: It would be better if we displayed the error message with toast.

**Review by Oliviier-dev — COMMENTED** [review]
Great work @Ke-bean, work on this minor change below:

**Review by andremugabo — DISMISSED** [review]
LGTM

**Review by Oliviier-dev — DISMISSED** [review]
LGTM

File: src/components/rootcomponents/MenuLinksIcons.tsx [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+
 interface MenuLinkIconProp {
   menuActive?: boolean;
   className?: string;
 }
 
 const MenuLinksIcons = ({ menuActive, className }: MenuLinkIconProp) => {
+  const navigate = useNavigate();
+  const handleClick = () => {
+    console.log('Hello world');
```
Comment (SergeInGithub) [inline src/components/rootcomponents/MenuLinksIcons.tsx]:
```suggestion
```

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Work on the review below and you're good

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Pull request merged**